Stripe/StripeiOSTests/STPIntentWithPreferencesTest.swift

```diff
@@ -93,19 +93,64 @@ class STPIntentWithPreferencesTest: XCTestCase {
         wait(for: [expectation], timeout: STPTestingNetworkRequestTimeout)
     }
 
-    func testRetrieveElementSession() {
+    func testRetrieveElementSession_deferredPayment() {
         let expectation = XCTestExpectation(description: "Retrieve ElementsSession")
         let client = STPAPIClient(publishableKey: STPTestingDefaultPublishableKey)
 
-        client.retrieveElementsSession { result in
+        let intentConfig = PaymentSheet.IntentConfiguration(mode: .payment(amount: 2000,
+                                                                           currency: "USD",
+                                                                           setupFutureUsage: .onSession),
+                                                            captureMethod: .automatic,
+                                                            paymentMethodTypes: ["card", "cashapp"])
+
+        client.retrieveElementsSession(withIntentConfig: intentConfig) { result in
             switch result {
-            case .success(let deferredIntent):
-                XCTAssertNotNil(deferredIntent)
-                XCTAssertEqual(deferredIntent.countryCode, "US")
-                XCTAssertNotNil(deferredIntent.linkSettings)
-                XCTAssertNotNil(deferredIntent.paymentMethodSpecs)
-                XCTAssertFalse(deferredIntent.orderedPaymentMethodTypes.isEmpty)
-                XCTAssertFalse(deferredIntent.paymentMethodTypes.isEmpty)
+            case .success(let elementsSession):
+                XCTAssertNotNil(elementsSession)
+                XCTAssertEqual(elementsSession.countryCode, "US")
+                XCTAssertNotNil(elementsSession.linkSettings)
+                XCTAssertNotNil(elementsSession.paymentMethodSpecs)
+                XCTAssertEqual(
```

Comment:
Since we can specify the payment methods now I think it makes sense to test these specifically.

Reply:
I'm not 100% sure whether or not the endpoint can change the provided payment method types fyi but this seems fine.